src/main/java/com/google/graphgeckos/dashboard/reader/BuildBotSetUpReader.java

```diff
@@ -0,0 +1,50 @@
+package com.google.graphgeckos.dashboard.reader;
+
+import com.google.api.client.util.Preconditions;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.lang.NonNull;
+
+public class BuildBotSetUpReader {
+  /** Path to the config file with the initial BuildBot data. */
+  private static String filePath = "src/main/resources/static/buildbots.txt";
+
+  private BuildBotSetUpReader() {}
+
+  private static List<BuildBotSetUpData> deserialize(String info) {
+    List<BuildBotSetUpData> bots = new ArrayList<>();
+    String[] splited = info.split("\n");
+    if (splited.length < 3 || splited.length % 3 != 0) {
+      throw new IllegalArgumentException("Every buildbot has a name, an initial id and a delay");
+    }
+
+    for (int i = 0; i < splited.length - 2; i += 3) {
+      bots.add(new BuildBotSetUpData(splited[i], Long.parseLong(splited[i + 1]), Long.parseLong(splited[i + 2])));
+    }
+    return bots;
+  }
+
+  /**
+   * Creates list of data that is needed for initializing and running instances of
+   * {@link com.google.graphgeckos.dashboard.fetchers.buildbot.BuildBotClient}.
+   *
+   * @return a list of data for initializing BuildBot fetchers.
+   */
+  public static List<BuildBotSetUpData> read() {
+    try {
+      return deserialize(Files.lines(Paths.get(filePath)).collect(Collectors.joining("\n")));
```

Comment:
I only skimmed this PR so far as Sam hasn't looked at the PR yet, but you could try to make sure to allow for cross-platform compatible line endings. I made a quick search and found this: https://intellipaat.com/community/32399/is-there-a-newline-constant-defined-in-java-like-environment-newline-in-c#:~:text=Java%20API%20%3A%20System.lineSeparator&text=It%20always%20returns%20the%20same,%22%5Cr%5Cn%22.

The potential issue with \n is that it assumes a UNIX style system.

Reply:
Done!

Reply:
You only replaced the line separator on a different part of the file, but not here where my comment is =).

Reply:
Ooops, sorry. Fixed it in the latest commit.